src/mesido/electricity_physics_mixin.py

```diff
@@ -178,6 +211,11 @@ def bounds(self):
         """
         bounds = super().bounds()
 
+        options = self.energy_system_options()
+        if options["electrolyzer_efficiency"] == ElectrolyzerOption.LINEARIZED_THREE_LINES_EQUALITY:
+            for line in self.__electrolyzer_is_active_linear_segment_bounds.keys():
+                bounds.update(self.__electrolyzer_is_active_linear_segment_bounds[line])
+
```

Comment:
See above, no need for check on the electrolyzer optionn and for the loop.

Reply:
Changed